api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,35 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  try {
+    const xToken = req.get('X-Access-Token');
+    if (!xToken) {
+      throw new ClientError(401, 'authentication required');
+    }
+    const payload = jwt.verify(xToken, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (error) {
+    console.error(error);
+  }
```

Comment:
Be sure to pass the `err` to `next()` so that Express can deal with it and send some kind of response to the client.